platform/ios/src/MGLScaleBar.mm

```diff
@@ -498,21 +464,21 @@ - (void)layoutSubviews {
     }
 
     // Re-layout the component bars and labels of the scale bar
-    CGFloat intrinsicContentHeight = self.intrinsicContentSize.height;
     CGFloat barWidth               = totalBarWidth/self.bars.count;
 
     BOOL RTL               = [self usesRightToLeftLayout];
     CGFloat halfLabelWidth = ceil(self.lastLabelWidth/2);
     CGFloat barOffset      = RTL ? halfLabelWidth : 0.0;
     
-    self.containerView.frame = CGRectMake(barOffset,
-                                          intrinsicContentHeight - MGLBarHeight,
+    CGFloat containerViewX = self.isOnScreenRight ? self.bounds.size.width - totalBarWidth - barOffset : barOffset;
```

Comment:
Does this also need the `halfLabelWidth` if on the right hand side?

Reply:
It needs the `barOffset`(calculated by `halfLabelWidth `) for layout direction right to left on either side. Otherwise, labels will display like this 👇
![image](https://user-images.githubusercontent.com/503040/96115686-ec6b8300-0f19-11eb-88ff-fe8a3c7272f7.png)